arangod/Aql/TraversalConditionFinder.cpp

```diff
@@ -729,21 +752,29 @@ bool TraversalConditionFinder::before(ExecutionNode* en) {
           case OptimizationCase::VERTEX:
           case OptimizationCase::EDGE: {
             // We have the Vertex or Edge variable in the statement
-            // Both have about the Same code and just differ in the used
-            // variable. auto usedVar = usedCase == OptimizationCase::VERTEX ?
-            // vertexVar : edgeVar;
-
-            auto conditionToOptimize = conditionWithInlineCalculations(
-                _plan, andNode->getMemberUnchecked(i - 1));
-
-            // Create a clone before we modify the Condition
-            AstNode* cloned = conditionToOptimize->clone(_plan->getAst());
-            // Retain original condition, as covered by this Node
-            coveredCondition->andCombine(cloned);
-            node->registerPostFilterCondition(conditionToOptimize);
+            AstNode* expr = andNode->getMemberUnchecked(i - 1);
+
+            // check if the filter condition can be executed on a DB server,
+            // deterministically
+            if (expr->isDeterministic() &&
+                (!ServerState::instance()->isRunningInCluster() ||
+                 expr->canRunOnDBServer(
+                     _plan->getAst()->query().vocbase().isOneShard()))) {
```

Comment:
Checks here are checking same things as in the method: 
```
case NODE_TYPE_FCALL: {
```

Maybe we we want to have a method + enum or something similar doing this.

Reply:
Still, minor issue. Code looks fine.